src/smt/set_defaults.cpp

```diff
@@ -109,37 +96,72 @@ void SetDefaults::setDefaultsPre(Options& opts)
     }
     opts.smt.unsatCoresMode = options::UnsatCoresMode::ASSUMPTIONS;
   }
-
-  if (opts.smt.checkProofs || opts.driver.dumpProofs)
+  // if check-proofs, dump-proofs, or proof-mode=full, then proofs being fully
+  // enabled is implied
+  if (opts.smt.checkProofs || opts.driver.dumpProofs
+      || opts.smt.proofMode == options::ProofMode::FULL)
   {
     opts.smt.produceProofs = true;
   }
 
-  if (opts.smt.produceProofs
-      && opts.smt.unsatCoresMode != options::UnsatCoresMode::FULL_PROOF)
+  // this check assumes the user has requested *full* proofs
```

Comment:
can't we add a check in this block that if the user requested proofs but set the proof mode to something other than full it's an error?

Reply:
Ah, indeed, that'd be nice.